frontend/app/src/components/history/ExternalTradeForm.vue

```diff
@@ -282,292 +289,316 @@ import { TaskType } from '@/types/task-type';
+    const notes = ref<string>('');
+    const type = ref<TradeType>('buy');
+
+    const errorMessages = ref<{ [field: string]: string[] }>({});
+
+    const quoteAmountInput = ref<any>(null);
+    const rateInput = ref<any>(null);
+
+    const baseRules = [
+      (v: string) =>
+        !!v || i18n.t('external_trade_form.validation.non_empty_base')
+    ];
+    const quoteRules = [
+      (v: string) =>
+        !!v || i18n.t('external_trade_form.validation.non_empty_quote')
+    ];
+    const amountRules = [
+      (v: string) =>
+        !!v || i18n.t('external_trade_form.validation.non_empty_amount')
+    ];
+    const rateRules = [
+      (v: string) =>
+        !!v || i18n.t('external_trade_form.validation.non_empty_rate')
+    ];
+    const quoteAmountRules = [
+      (v: string) =>
+        !!v || i18n.t('external_trade_form.validation.non_empty_quote_amount')
+    ];
+
+    const baseHint = computed<string>(() => {
+      return unref(type) === 'buy'
+        ? i18n.t('external_trade_form.buy_base').toString()
+        : i18n.t('external_trade_form.sell_base').toString();
+    });
 
-  get quoteHint(): string {
-    return this.type === 'buy'
-      ? this.$t('external_trade_form.buy_quote').toString()
-      : this.$t('external_trade_form.sell_quote').toString();
-  }
+    const quoteHint = computed<string>(() => {
+      return unref(type) === 'buy'
+        ? i18n.t('external_trade_form.buy_quote').toString()
+        : i18n.t('external_trade_form.sell_quote').toString();
+    });
 
-  get shouldRenderSummary(): boolean {
-    return !!(this.type && this.base && this.quote && this.amount && this.rate);
-  }
+    const shouldRenderSummary = computed<boolean>(() => {
+      return !!(
+        unref(type) &&
+        unref(base) &&
+        unref(quote) &&
+        unref(amount) &&
+        unref(rate)
+      );
+    });
 
-  rateMessages: string[] = [];
-
-  id: string = '';
-  datetime: string = '';
-  amount: string = '';
-  rate: string = '';
-  quoteAmount: string = '';
-  selectedCalculationInput: 'rate' | 'quoteAmount' = 'rate';
-  fee: string = '';
-  feeCurrency: string = '';
-  link: string = '';
-  notes: string = '';
-  type: TradeType = 'buy';
-
-  get fetching(): boolean {
-    return this.isTaskRunning(TaskType.FETCH_HISTORIC_PRICE).value;
-  }
+    const fetching = computed<boolean>(() => {
+      return isTaskRunning(TaskType.FETCH_HISTORIC_PRICE).value;
+    });
 
-  get numericAmount(): BigNumber {
-    return bigNumberify(this.amount);
-  }
+    const numericAmount = computed<BigNumber>(() => {
+      return bigNumberify(unref(amount));
+    });
 
-  get numericRate(): BigNumber {
-    return bigNumberify(this.rate);
-  }
+    const numericRate = computed<BigNumber>(() => {
+      return bigNumberify(unref(rate));
+    });
 
-  get numeridFee(): BigNumber {
-    return bigNumberify(this.fee);
-  }
+    const numericFee = computed<BigNumber>(() => {
+      return bigNumberify(unref(fee));
+    });
 
```

Comment:
do you think it would make sense to extract a common function here? since this is repeated? one that would take a `Ref<string>` and return a `ComputedRef<BigNumber>`?